pkgs/os-specific/linux/apparmor/default.nix

```diff
@@ -249,8 +269,35 @@ let
     meta = apparmor-meta "kernel patches";
   };
 
+  # Generate generic AppArmor rules in a file,
+  # from the closure of given rootPaths.
+  # To be included in an AppArmor profile like so:
+  # include "$(apparmorRulesFromClosure {} [pkgs.hello]}"
+  apparmorRulesFromClosure =
```

Comment:
We should have a test for this if we offer it as an API to create closures. Doesn't have to be huge but should test that
* We do not include everything
* We do include what was passed in (for etc, share, lib, ...)

Reply:
I've added a `nixos/tests/apparmor.nix` testing `apparmorRulesFromClosure`, by:
1. prepending a string of the relevant packages' name and version on each line,
2. sorting the lines according to these prepended strings,
3. removing the prepended strings,
4. using `diff` against the expected output.